pkg/core/dao/dao.go

```diff
@@ -585,6 +585,13 @@ func (dao *Simple) DeleteBlock(h util.Uint256, w *io.BufBinWriter) error {
 	for _, tx := range b.Transactions {
 		copy(key[1:], tx.Hash().BytesBE())
 		batch.Delete(key)
+		if p2pSigExtensions {
```

Comment:
@roman-khimov, I see a problem here: if `P2PStateExchangeExtensions` are on, then blocks are removed up to `H-StateSuncInterval-MaxTraceableBlocks` height, where H is the current chain's height (which is the same as the latest state synchronisation point, see the [code](https://github.com/nspcc-dev/neo-go/blob/master/pkg/core/blockchain.go#L750)). But if we're to remove conflicting transactions up to the same height, then the behaviour of archival and non-archival nodes may differ, because non-archival recently restored node has conflicting transactions only up to `H-MaxTraceableBlocks` heigh.

So I think that we need to remove conflicting transactions up to `H-MaxTraceableBlocks` respectively to the  `P2PStateExchangeExtensions` setting. What do you think?

Reply:
What's the practical difference? Conflicting transaction is supposed to invalid at `H` anyway because of VUB. Whether we store dummies for `H-SSI-MTB` or `H-MTB` doesn't change anything.